src/bus/mod.rs

```diff
@@ -164,8 +164,19 @@ impl Bus {
 
                 addr
             }
-            UnixSocket::Dir(_) => bail!("`dir` transport is not supported (yet)."),
-            UnixSocket::TmpDir(_) => bail!("`tmpdir` transport is not supported (yet)."),
+            UnixSocket::Dir(dir) | UnixSocket::TmpDir(dir) => {
+                // for tmpdir, the server _may_ create an abstract socket but isn't required to do so
+                // https://dbus.freedesktop.org/doc/dbus-specification.html
+                // so our tmpdir is the same as dir, as the specification suggests
```

Comment:
We don't need to comment about this here (should go in the commit log though). We don't want abstract socket because nobody really uses them anymore. :)